app/src/main/java/org/oppia/app/home/topiclist/TopicListAdapter.kt

```diff
@@ -107,34 +119,67 @@ class TopicListAdapter(
     return itemList.size
   }
 
-  private class WelcomeViewHolder(
-    val binding: WelcomeBinding
-  ) : RecyclerView.ViewHolder(binding.root) {
+  private class WelcomeViewHolder(val binding: WelcomeBinding) : RecyclerView.ViewHolder(binding.root) {
     internal fun bind(userAppHistoryViewModel: UserAppHistoryViewModel) {
       binding.viewModel = userAppHistoryViewModel
     }
   }
 
-  private class PromotedStoryViewHolder(
-    val binding: PromotedStoryCardBinding
-  ) : RecyclerView.ViewHolder(binding.root) {
-    internal fun bind(promotedStoryViewModel: PromotedStoryViewModel) {
-      binding.viewModel = promotedStoryViewModel
+  inner class PromotedStoryListViewHolder(val binding: PromotedStoryListBinding) :
+    RecyclerView.ViewHolder(binding.root) {
+    internal fun bind(
+      activity: AppCompatActivity,
+      promotedStoryListViewModel: PromotedStoryListViewModel,
+      promotedStoryList: MutableList<PromotedStoryViewModel>
+    ) {
+      binding.viewModel = promotedStoryListViewModel
+      val promotedStoryAdapter = PromotedStoryListAdapter(promotedStoryList)
+      val horizontalLayoutManager = LinearLayoutManager(activity, LinearLayoutManager.HORIZONTAL, false)
+      binding.promotedStoryListRecyclerView.apply {
+        layoutManager = horizontalLayoutManager
+        adapter = promotedStoryAdapter
+      }
+
+      val snapHelper = PagerSnapHelper()
```

Comment:
Is this the desired behavior per the mocks? I assumed we wanted smooth scrolling between the items, not snapping. Or does it only snap when the user lifts their finger after scrolling through the list?

Reply:
@BenHenning As per my conversation with @seanlip and @mschanteltc the items should snap to middle and not smooth scroll. And yes, it will only snap when the finger is lift up.

Reply:
Cool, that seems nice then. Thanks for the context!

Note that it might be worth dropping a brief line comment here explaining why we want the snapping & what the expected behavior should be since this is really hard to verify with automated tests (if it ever stopped working in the future, this comment helps describe what the code intends to do which narrows down what might have broken for whoever is investigating).

Reply:
Added comment:
`The PagerSnapHelper is used to snap between items rather than smooth scrolling,so that the item is completely visible in [HomeFragment].`